source/blog/2017-02-13-automate-building-atomic-host.html.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Automate Building Atomic Host
+author: trishnag
+date: 2017-02-13 12:00:00 UTC
+published: true
+comments: true
+tags: atomic, fedora, ansible, automation, ostree
+---
+
+[Project Atomic](http://www.projectatomic.io/) hosts are built from standard RPM packages which have been composed into filesystem trees using rpm-ostree. This guide provides method for automation of Building Atomic host(Creating new trees).
+
+One of the primary benefits to Atomic Host and OSTree has been the ability to "configure once, deploy many times" using custom OSTree images. But the process for doing so wasn't streamlined or well-documented. I'm helping change that. I'm going to describe how to build atomic host in automated way. At the end of the article you'll be able to create VM from QCOW2 image which is going to boot in to the own OSTree and the VM can also be used for testing Atomic host, please feel free to open issue in that case: [https://pagure.io/atomic-wg/issues](https://pagure.io/atomic-wg/issues).
+
+The procedure and playbook below will enable you to create your own Atomic Host OSTree image. This is the first step in creating your own "distributions" of Atomic Host to install on your cloud servers. Note that it will install a bunch of requirements on your local server, as well as using system resources heavily. As such, you may want to run it on a development machine instead of your personal laptop.
+
+There is also Documentation on Project Atomic which describes about Compose own Tree and Build Atomic host: [http://www.projectatomic.io/docs/compose-your-own-tree](http://www.projectatomic.io/docs/compose-your-own-tree/).
+
+
+# Requirements
+* Fedora Atomic QCOW2 Image. The image can be downloaded from here: [Fedora-Atomic](https://getfedora.org/en/atomic/download/)
+* Make sure that Ansible is installed on your system: `$ sudo dnf install ansible`. Note that I am installing on my [Fedora-Workstation](https://getfedora.org/en/workstation/download).
+
+
+## Process
+Clone the Git repo on your working machine [Build-Atomic-Host](https://github.com/trishnaguha/build-atomic-host/)
+
+```
+$ git clone https://github.com/trishnaguha/build-atomic-host.git
+$ cd build-atomic-host
+```
+
+
+## Create VM from the QCOW2 Image
+The following creates VM from QCOW2 Image where username is `atomic-user` and password is `atomic`.
+Here `atomic-node` in the instance name.
+
+```
+$ sudo sh create-vm.sh atomic-node /path/to/fedora-atomic25.qcow2
+# For example: /var/lib/libvirt/images/Fedora-Atomic-25-20170131.0.x86_64.qcow2
+```
+
+
+## Install Requirements and Start HTTP Server
+The tree is made available via web server. The following playbook installs requirements, creates directory structure, initializes OSTree repository and starts the HTTP server.
+
+```
+$ ansible-playbook setup.yml --ask-sudo-pass
+```
+Use `ip addr` to check IP Address of the HTTP server.
```

Comment:
it's not clear which IP address users should use here.  Shouldn't you just recommend 127.0.0.1?  If not, which one should they use?